agent/api/task.go

```diff
@@ -171,14 +171,19 @@ func (task *Task) initializeEmptyVolumes() {
 				continue
 			}
 			if _, ok := vol.(*EmptyHostVolume); ok {
-				if container.TransitionDependencySet.ContainerDependencies == nil {
-					container.TransitionDependencySet.ContainerDependencies = make([]ContainerDependency, 0)
+				if container.TransitionDependenciesMap == nil {
+					container.TransitionDependenciesMap = make(map[ContainerStatus]TransitionDependencySet)
 				}
-				container.TransitionDependencySet.ContainerDependencies = append(container.TransitionDependencySet.ContainerDependencies, ContainerDependency{
+				contDep := ContainerDependency{
 					ContainerName:   emptyHostVolumeName,
 					SatisfiedStatus: ContainerRunning,
-					DependentStatus: ContainerCreated,
-				})
+				}
+				if _, ok := container.TransitionDependenciesMap[ContainerCreated]; !ok {
+					container.TransitionDependenciesMap[ContainerCreated] = TransitionDependencySet{}
+				}
+				deps := container.TransitionDependenciesMap[ContainerCreated]
+				deps.ContainerDependencies = append(deps.ContainerDependencies, contDep)
```

Comment:
I think you need a nil check for `deps.ContainerDependencies`?

Reply:
Looks like we might not? 
https://play.golang.org/p/1v0JJSknXQX